md-dao/src/main/java/com/wipro/ats/bdre/md/dao/UserRolesDAO.java

```diff
@@ -49,6 +52,27 @@
         return userRoles;
     }
 
+    public Map<Integer,String> diffRoleList() {
+        Session session = sessionFactory.openSession();
+        session.beginTransaction();
+        Criteria criteria = session.createCriteria(UserRoles.class);
+        ProjectionList p1=Projections.projectionList();
+        p1.add(Projections.property("role"));
+        p1.add(Projections.property("userRoleId"));
+        criteria.setProjection(p1);
+        List l=criteria.list();
+        Iterator it=l.iterator();
+        Map<Integer,String> diffRoles=new HashMap<>();
+        while(it.hasNext())
+        {
+            Object ob[] = (Object[])it.next();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Move the array designator from the variable to the type. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1197)